src/tiff.rs

```diff
@@ -132,9 +194,15 @@ pub struct TiffEncoder<W> {
 
 // Utility to simplify and deduplicate error handling during 16-bit encoding.
 fn u8_slice_as_u16(buf: &[u8]) -> ImageResult<&[u16]> {
-    bytemuck::try_cast_slice(buf)
+    bytemuck::try_cast_slice(buf).map_err(|err| {
         // If the buffer is not aligned or the correct length for a u16 slice, err.
-        .map_err(|_| ImageError::IoError(std::io::ErrorKind::InvalidData.into()))
+        //
+        // `bytemuck::PodCastError` of bytemuck-1.2.0 does nto implement
+        // `Error` and `Display trait.
```

Comment:
Reported upstream: https://github.com/Lokathor/bytemuck/issues/22

Reply:
Added a link to the issue (and fixed typo).